A32-xds-circuit-breaking.md

```diff
@@ -0,0 +1,137 @@
+routing](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_routing#arch-overview-http-routing-priority), 
+requests to a single upstream cluster conform to the single circuit breaking
+configuration.
+
+Envoy handles HTTP traffic with [connection pooling](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/connection_pooling#connection-pooling),
+and many circuit breaking parameters are configured for
+connection/connection pool management. Since each gRPC client creates at most
+one HTTP/2 connection to a single upstream endpoint address, those
+parameters do not apply to gRPC and will be ignored. See more in
+[Other parameters considered](#other-parameters-considered).
+
+With the introduction of [gRFC A28: xDS Traffic Splitting and Routing](https://github.com/grpc/proposal/blob/master/A28-xds-traffic-splitting-and-routing.md), 
+it became possible for there to be more than one CDS LB policy instance for a 
+given cluster (e.g., if one route pointed to a given cluster and a different 
+route split traffic between that cluster and other clusters).  However, after 
+implementing the changes described in [gRPC A31: xDS RouteActions Support](https://github.com/grpc/proposal/pull/192),
+there will be only one CDS LB policy instance for a given cluster, which will
+make the CDS LB policy the logical place to implement circuit breakers.
+
+## Overview
+
+The `Cluster` resource encodes the circuit breaking parameters in a list of
+[Thresholds](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cluster/circuit_breaker.proto#cluster-circuitbreakers-thresholds)
+messages, where each message specifies the limits for a particular 
+RoutingPriority. gRPC will look only at the first entry in the list for 
+priority [`DEFAULT`](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/core/base.proto#enum-core-routingpriority).
+
+Of the parameters in the `Thresholds` message, gRPC will support only
+`max_requests`, which sets the maximum number of requests that can be in flight
+to the cluster at any given time.  None of the other parameters are applicable
+to gRPC; for details, see [Other parameters considered](#other-parameters-considered).
+The design and implementation parts of this doc will mainly focus on the
+application of the `max_requests` circuit breaking parameter in gRPC.
+
+To match Envoy's implementation, circuit breaking parameters will always be 
+enforced with certain defaults, even if the gRPC client receives no applicable 
+circuit breakers. Users can effectively turn circuit breaking off by setting
+thresholds to very high values, for example, to 
+`std::numeric_limits<uint32_t>::max()`.
+
+## Detailed design
+
+### Maintaining thresholds
+
+Each CDS LB policy maintains a threshold for the maximum number of outstanding 
+requests to hosts in the cluster that this policy is load balancing for. This
+threshold will be updated when receiving an update from the XdsClient cluster 
```

Comment:
s/threshold/limit/

Reply:
Fixed, along with a couple of other "threshold" and changed them to "limit" for consistency.